contracts/lib/OrderValidator.sol

```diff
@@ -280,7 +279,7 @@ contract OrderValidator is Executor, ZoneInteraction {
                 // Update order status and fill amount, packing struct values.
                 _orderStatus[orderHash].isValidated = true;
                 _orderStatus[orderHash].isCancelled = false;
-                _orderStatus[orderHash].numerator = uint120(filledNumerator);
+                _orderStatus[orderHash].numerator = uint120(numerator);
                 _orderStatus[orderHash].denominator = uint120(denominator);
             }
```

Comment:
After this PR these two blocks could be moved outside of the if statement and reduced to a single one, as they are identical.